shell/browser/native_window_mac.mm

```diff
@@ -1810,6 +1812,27 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
   [window_ setAcceptsMouseMovedEvents:forward];
 }
 
+gfx::Rect NativeWindowMac::GetWindowControlsOverlayRect() {
+  gfx::Rect bounding_rect;
+  if (!has_frame() && buttons_view_ && ![buttons_view_ isHidden] &&
+      !traffic_light_position_) {
```

Comment:
The Window Control Overlay should be enabled for custom traffic light position.

Reply:
@zcbenz as I mentioned in https://github.com/electron/electron/pull/29253#discussion_r640873735, the problem with a custom traffic light position and Window Control overlay is that it is indeterminate what the Windows Control Overlay region should be.  For example if someone for some reason moves the traffic light position out of the title bar area then the Windows Control Overlay region would be the whole title bar area.  For example if some reason someone wanted the traffic lights on the bottom of the screen should the Windows Control Overlay be a really large rectangle?

Reply:
Yes, if user uses an unusual `trafficLightPosition` then it is fine for Windows Control Overlay to be a really large rect.

Windows Control Overlay is an optional feature, if user does not use it then the size of it does not matter, if user wants to use it then it is their responsibility to provide a reasonable `trafficLightPosition` (or it is API designer's responsibility to take unusual `trafficLightPosition` into consideration).

For example when using `backgroundColor` option, the `transparent` option will not work correctly if the specified color is an opaque color. But that does not mean we have to disable `transparent` option when `backgroundColor` is used, it is user's responsibility to set a reasonable `backgroundColor` to make `transparent` work.

Reply:
ok sounds good